Demo/app/src/main/java/com/juanocampo/swipeable/swipeablelist/swpeable/adapter/SwipeableHelperAdapter.java

```diff
@@ -17,40 +20,39 @@
 
 public abstract class SwipeableHelperAdapter extends RecyclerView.Adapter {
 
+
     public interface SwipeAdapterActions {
         void swiped(int position);
     }
 
     private final SwipeableHelperCallback callback;
     private SwipeAdapterActions listener;
     private List<RecyclerView.ViewHolder> swipeableViewHolders;
+    private final LinearLayoutManager manager;
 
 
-    public SwipeableHelperAdapter(SwipeAdapterActions listener, RecyclerView recyclerView) {
-        callback = new SwipeableHelperCallback(this);
-        ItemTouchHelper mItemTouchHelper = new ItemTouchHelper(callback);
-        mItemTouchHelper.attachToRecyclerView(recyclerView);
+    public SwipeableHelperAdapter(SwipeAdapterActions listener, RecyclerView recyclerView, LinearLayoutManager manager) {
+        callback = new SwipeableHelperCallback(this, recyclerView);
         this.listener = listener;
         this.swipeableViewHolders = new ArrayList<>();
+        this.manager = manager;
+        attachToRecycler(recyclerView);
+    }
+
+    public void attachToRecycler(RecyclerView recyclerView) {
+        ItemTouchHelper mItemTouchHelper = new ItemTouchHelper(callback);
+        mItemTouchHelper.attachToRecyclerView(recyclerView);
     }
 
+
+
     @Override
     public final RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
         RecyclerView.ViewHolder itemViewHolder = onCreateSwipeViewHolder(parent, viewType);
         swipeableViewHolders.add(itemViewHolder);
```

Comment:
We will need to store the view holders in order to access them when we need to manipulate them. 
For the case you are concern, i already have this in mind and we can just simple avoid it storing the  selected visible position, with that if the list decides re-user the viewHolder or simple recycle them, we will just fire the animation to the visible position. Please check this line
https://github.com/ingjuanocampo/SwipeableRecyclerView/pull/1#discussion-diff-74665731R80